### PR TITLE
Improved CML XSD loading using an entity resolver. Also use try-with-…

### PR DESCRIPTION
**Pull request by johnmay**
…resources to ensure all streams are closed correctly.

### CODE REVIEW

**Comment by egonw** [comment]
Thanks. Sorry I could not find the time this weekend to look at this issue.

**Comment by johnmay** [comment]
Thanks

File: app/depict/src/test/java/org/openscience/cdk/depict/DepictionTest.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public void depictAsSvg() throws CDKException {
         SmilesParser sp = new SmilesParser(SilentChemObjectBuilder.getInstance());
         IAtomContainer ac = sp.parseSmiles("[nH]1cccc1");
         String svg = dg.depict(ac).toSvgStr();
-        String nl = System.getProperty("line.separator");
+        String nl = "\n";
```
Comment (egonw) [inline app/depict/src/test/java/org/openscience/cdk/depict/DepictionTest.java:78]: Mmm... this was probably there for a reason.

Comment (uli-f) [reply app/depict/src/test/java/org/openscience/cdk/depict/DepictionTest.java:78]: Well, given that most (all?) of the development and testing happens on linux/mac machines and the test broke an my windows machine before switching **to** `\n` it seemed the better choice to give up on the assumed platform indepdence of the line separator 🤷🏼

**Review by egonw — APPROVED** [review]
Okay, I looked at it from various sides and had to get into XML parsing again. But I cannot see any problems and suggest to merge.

**Comment by uli-f** [comment]
@egonw Thanks for merging. Feels good to have made another (small) contribution 😃

**Pull request merged**